### PR TITLE
[OLS-585] added tolerations and nodeselector to olsconfig CR

### PR DESCRIPTION
**Pull request by xrajesh**
## Description

OLSConfig CRD provides option for all its deployments ability to choose NodeSelector and Tolerations

## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue [OLS-585](https://issues.redhat.com/browse/OLS-585)
- Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

File: internal/controller/utils.go [hunk 1]
```diff
@@ -45,6 +45,22 @@ func setDeploymentReplicas(deployment *appsv1.Deployment, replicas int32) bool {
 	return false
 }
 
+func setTolerations(deployment *appsv1.Deployment, tolerations []corev1.Toleration) bool {
+	if !apiequality.Semantic.DeepEqual(deployment.Spec.Template.Spec.Tolerations, tolerations) {
```
Comment (tisnik) [inline internal/controller/utils.go:49]: should the ! (negation) be there?

File: internal/controller/utils.go [hunk 2]
```diff
@@ -45,6 +45,22 @@ func setDeploymentReplicas(deployment *appsv1.Deployment, replicas int32) bool {
...
+		deployment.Spec.Template.Spec.Tolerations = tolerations
+		return true
+	}
+	return false
+}
+
+func setNodeSelector(deployment *appsv1.Deployment, nodeSelector map[string]string) bool {
+	if !apiequality.Semantic.DeepEqual(deployment.Spec.Template.Spec.NodeSelector, nodeSelector) {
```
Comment (tisnik) [inline internal/controller/utils.go:57]: dtto

**Review by tisnik — COMMENTED** [review]
it looks ok in overall. I'm just not sure about the semantic equality comparison with negation

File: internal/controller/utils.go (hunk 2 above)
Comment (xrajesh) [reply internal/controller/utils.go:57]: Yes , the object is different than what is present , we update.

File: internal/controller/utils.go (hunk 1 above)
Comment (xrajesh) [reply internal/controller/utils.go:49]: Yes , the object is different than what is present , we update.

File: bundle/manifests/lightspeed-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -562,8 +578,8 @@ spec:
   version: 0.0.1
   relatedImages:
     - name: lightspeed-service-api
-      image: quay.io/redhat-user-workloads/crt-nshift-lightspeed-tenant/ols/lightspeed-service@sha256:b8e0bfe7e4a241885bcd30baa7066239276d5a6c7de8b6526d4861c6d3d4bfe7
+      image: quay.io/openshift-lightspeed/lightspeed-service-api:latest
     - name: lightspeed-console-plugin
-      image: quay.io/redhat-user-workloads/crt-nshift-lightspeed-tenant/ols/lightspeed-console@sha256:069828c27813e618dda86a498946d06d27daf73a8ff8d601a3cb72d00324a900
+      image: quay.io/openshift-lightspeed/lightspeed-console-plugin:latest
```
Comment (raptorsun) [inline bundle/manifests/lightspeed-operator.clusterserviceversion.yaml:583]: need to set them back to konflux images manually.
I will update the script later.

File: internal/controller/ols_console_reconciliator_test.go [hunk 4]
```diff
@@ -199,6 +199,61 @@ var _ = Describe("Console UI reconciliator", Ordered, func() {
...
+		}
+
+		By("Reconcile the app server")
+		err = reconciler.reconcileConsoleUIDeployment(ctx, olsConfig)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Get the deployment")
+		err = k8sClient.Get(ctx, types.NamespacedName{Name: ConsoleUIDeploymentName, Namespace: OLSNamespaceDefault}, dep)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(dep.Spec.Template.Spec.Tolerations).NotTo(BeNil())
```
Comment (raptorsun) [inline internal/controller/ols_console_reconciliator_test.go:228]: this check seems unneccesary

Comment (xrajesh) [reply internal/controller/ols_console_reconciliator_test.go:228]: removed both the nil check.

**Pull request merged**